docs/user_guide/system_architecture.rst

```diff
@@ -0,0 +1,190 @@
+may implement their own Overseer based on their deployment environment.  In that case, users also need to implement
+their own Overseer Agent.
+
+:class:`Overseer Agent Spec<nvflare.apis.overseer_spec.OverseerAgent>`
+
+NVIDIA FLARE provides two implementations:
+
+    - :class:`HttpOverseerAgent<nvflare.ha.overseer_agent.HttpOverseerAgent>` to work with the Overseer server. For NVIDIA
+      FLARE 2.1.0, the provisioning tool will automatically map parameters specified in Overseer into the arguments for
+      the HttpOverseerAgent.
+    - :class:`DummyOverseerAgent<nvflare.ha.dummy_overseer_agent.DummyOverseerAgent>` is a dummy agent that simply
+      returns the configured endpoint as the hot FL server. The dummy agent is used when a single FL server is configured
+      and no Overseer server is necessary in an NVIDIA FLARE system. When DummyOverseerAgent is specified, the provisioning
+      tool will include all arguments into the Overseer Agent section of generated json files.
+
+Job Definition Manager
+----------------------
+The Job Definition Manager config specifies the Python object that manages the access and manipulation of Job Definition objects stored in the Job Storage.
+
+The system reserved component id, job_manager, is used to denote the Job Definition Manager in the project.yml file.
+
+This component is specified as one item in the components.server section.
+
+This configuration is included in the fed_server.json of the Server’s Startup Kit.
+
+:class:`Job Definition Manager Spec<nvflare.apis.job_def_manager_spec.JobDefManagerSpec>`
+
+NVIDIA FLARE provides a simple implementation that is based on scanning of job definition objects:
+
+    - :class:`Simple Job Def Manager<nvflare.apis.impl.job_def_manager.SimpleJobDefManager>`
+
+Job Scheduler
+-------------
+The Job scheduler is responsible for determining the next job to run. Job scheduler config specifies the Job scheduler Python object.
+
+The system reserved component id, job_scheduler, is used to denote the Job Scheduler in the project.yml file.
+
+This component is specified as one item in the components.server section.
+
+This configuration is included in the fed_server.json of the Server’s Startup Kit.
+
+:class:`Job Scheduler Spec<nvflare.apis.job_scheduler_spec.JobSchedulerSpec>`
+
+NVIDIA FLARE provides a default implementation of the Job Scheduler that does resource based scheduling as described in the beginning:
+
+    - :class:`Default Job Scheduler<nvflare.app_common.job_schedulers.job_scheduler.DefaultJobScheduler>`
+
+Storage
+-------
+
+Study Storage
+^^^^^^^^^^^^^
+The Study definition is stored in a persistent store. The Study Storage config specifies the Python object that manages the access to the store.
+
+This component is specified as one item in the components.server section.
+
+This configuration is included in the fed_server.json of the Server’s Startup Kit.
```

Comment:
I think it might be better to put Study Storage below or inside StudyManager.
Since it is required by StudyManager?